analytics.js

```diff
@@ -276,6 +282,20 @@ class Analytics {
     }
   }
 
+  /**
+   * Prepares the data for integrationsObj
+   *
+   * @param {*} integration
+   * @param {*} integrationInstance
+   * @memberof Analytics
+   */
+  prepareDataForIntegrationsObj(integration, integrationInstance) {
+    if (integration.name === "GA4" && integrationInstance.isReady()) {
+      this.integrationsData[integration.name] =
+        integrationInstance.getDataForIntegrationsObject();
+    }
```

Comment:
There is no need to be explicit for GA4. You should do this for all integrations.

Rough logic:
```
integrationInstances.forEach((intgInstance) => {
  if (intgInstance["getDataForIntegrationsObject"]) {
    // As each integration will have a unique display name, there is no need for deep object merge
    this.integrationsData = { ...this.integrationsData, ...intgInstance["getDataForIntegrationsObject"]() };
  }
});
```

Reply:
updated